crowbar_framework/app/models/ntp_service.rb

```diff
@@ -110,8 +110,13 @@ def skip_unchanged_node?(node_name, old_role, new_role)
     # if the node changed roles, then we need to apply
     return false if node_changed_roles?(node_name, old_role, new_role)
 
-    # if attributes have changed, we need to run
-    return false if node_changed_attributes?(node_name, old_role, new_role)
+    # if attributes have changed except external_servers, we need to apply
+    return false if relevant_attributes_changed_if_roles?(node_name, old_role, new_role,
+      ["external_servers"], ["ntp-client"])
+
+    # if the node is a client and the servers has changed, we need to apply
+    return false if new_role.elements["ntp-client"].include?(node_name) &&
```

Comment:
(readding my comment that is hidden now)

No need for the `new_role.elements["ntp-client"].include?(node_name)` part of the test: if there's a change in ntp-server nodes, then we need to apply everywhere.